src/main/java/duke/subfiles/TaskList.java

```diff
@@ -0,0 +1,396 @@
+                if (d.getDateAsLocalDate().equals(date)) {
+                    deadlines.add(d);
+                }
+            }
+        }
+
+        if (deadlines.size() == 0) {
+            output = new StringBuilder("You have no deadlines due on " + date.toString() + ".");
+        } else {
+            int i = 1;
+            output = new StringBuilder("Here are the deadlines due on " + date.toString() + ":");
+            for (Deadline d : deadlines) {
+                output.append("\n").append(i).append(". ").append(d.toString());
+                i++;
+            }
+        }
+
+        System.out.println(output);
+        return output.toString();
+    }
+
+    /**
+     * Prints the list of events added by the user till this point,
+     * happening on the date specified by the user, based on the order
+     * they were added by the user.
+     *
+     * @param date The date specified by the user.
+     */
+    private String printEventsOnDate(LocalDate date) {
+        StringBuilder output;
+        ArrayList<Event> events = new ArrayList<>();
+
+        for (Task t : tasks) {
+            if (t instanceof Event) {
+                Event e = (Event) t;
+                if (e.getDateAsLocalDate().equals(date)) {
+                    events.add(e);
+                }
+            }
+        }
+
+        if (events.size() == 0) {
+            output = new StringBuilder("You have no events due on " + date.toString() + ".");
+        } else {
+            int i = 1;
+            output = new StringBuilder("Here are the events due on " + date.toString() + ":");
+            for (Event e : events) {
+                output.append("\n").append(i).append(". ").append(e.toString());
+                i++;
+            }
+        }
+
+        System.out.println(output);
+        return output.toString();
+    }
+
+    /**
+     * Prints the list of deadlines and events added by the user till this point,
+     * due or happening on the date specified by the user, based on the order
+     * they were added by the user.
+     *
+     * @param s The date specified by the user.
+     * @throws DateFormatException If the specified date is incorrectly
+     *                             formatted.
+     */
+    public String printTasksOnDate(String s) throws DateFormatException {
+        try {
+            LocalDate date = LocalDate.parse(s);
+            String outputUpper = printDeadlinesOnDate(date);
+            String outputLower = printEventsOnDate(date);
+            return outputUpper + "\n" + outputLower;
+        } catch (DateTimeParseException e) {
+            throw new DateFormatException();
+        }
+    }
+
+    /**
+     * Prints the list of tasks added by the user till this point,
+     * containing the keyword specified by the user, based on the order
+     * they were added by the user.
+     *
+     * @param s User input containing the keyword.
+     */
+    public String findTasksWithKeyword(String s) {

```

Comment:
Can consider changing parameter name to something more meaningful, e.g. "keyword" 😄